src/java.base/windows/native/libnio/ch/UnixDomainSockets.c

```diff
@@ -23,6 +23,10 @@
  * questions.
  */
 
+#ifndef UNICODE
+#define UNICODE 1
+#endif
```

Comment:
Setting UNICODE has wider impact, can you explain why you are adding this? Does it go away if WSASocketW is used instead?

Reply:
That was why I only set it in this file, but using WSASocketW and WSAEnumProtocolsW (and the corresponding W data types) seems to work.

Reply:
Concerning the general question about supporting the MKS provider. It has different behavior. For instance the filesystem node created, appears to be a regular file rather than a reparse point (a la Microsoft). We would have to think carefully about these differences if we wanted to support it.